frankenphp.go

```diff
@@ -172,9 +172,9 @@ func NewRequestWithContext(r *http.Request, opts ...RequestOption) (*http.Reques
 		fc.logger = getLogger()
 	}
 
-	if splitPos := splitPos(fc, r.URL.Path); splitPos > -1 {
-		fc.docURI = r.URL.Path[:splitPos]
-		fc.pathInfo = r.URL.Path[splitPos:]
+	if splitPos := splitPos(fc, r.URL.EscapedPath()); splitPos > -1 {
```

Comment:
You could store the result of `EscapedPath()` in a variable to not call the function several times.